packages/gluestick/src/commands/compareVersions/compareModuleVersions.js

```diff
@@ -0,0 +1,71 @@
+
+const semver = require('semver');
+const chalk = require('chalk');
+
+const packageName = '/package.json';
+
+/**
+ * Opens the project's package.json file and compares version requirements to
+ * the actual version of each dependency and devDependency. It lists the
+ * node_modules that don't satisfy the requirement or are missing and suggests
+ * running `npm install` to amend the issue.
+ *
+ * Corner cases: there are no dependencies and/or devDependencies, the version
+ * is a path to a file, the input file to the package.json or node_modules dir
+ * is either undefined or incorrect.
+ *
+ * Return - returns array of dependencies that are missing or out of date.
+ */
+const compareModuleVersions = (projectPackage, modulePath) => {
+  const discrepancies = [];
+  let devDepDifference = false;
+  let allDependencies = projectPackage;
+
+  if (projectPackage.dependencies === undefined || projectPackage.devDependencies === undefined) {
+    if (projectPackage.dependencies === undefined || projectPackage.devDependencies === undefined) {
+      return discrepancies;
+    }
+    allDependencies = projectPackage.dependencies || projectPackage.devDependencies;
+  } else {
+    const totalNumDependencies = Object.keys(projectPackage.dependencies).length +
+      Object.keys(projectPackage.devDependencies).length;
+    allDependencies = Object.assign(projectPackage.dependencies, projectPackage.devDependencies);
+    if (Object.keys(allDependencies).length < totalNumDependencies) {
+      devDepDifference = true;
+    }
+  }
+
+  Object.keys(allDependencies).forEach(dep => {
+    const tempPath = modulePath.concat('/', dep, packageName);
+    let tempPackage = true;
+    try {
+      tempPackage = require(tempPath);
+    } catch (e) {
+      console.log(`${chalk.red('The node_module ')}`, `${chalk.yellow(dep)}`, `${chalk.red(' does not exist')}`);
+      const name = ' '.concat(dep);
+      discrepancies.push(name);
+      return;
+    }
+
+    if (!tempPackage || !semver.satisfies(tempPackage.version, allDependencies[dep])) {
+      const fileVersion = allDependencies[dep].match(/\d+\.\d+\.\d+/)[0];
```

Comment:
It might work for this case, but it's not like this all the time. For instance I frequently have
```
"gluestick": "../gluestick/packages/gluestick",
```
I use that to when testing gluestick on an internal project. So the code above will fail on this. Moreover, it's more common for ppl when using relative/absolute paths not to have any version string in it.